packages/@headlessui-react/src/utils/render.ts

```diff
@@ -57,7 +57,16 @@ export type PropsForFeatures<T extends RenderFeatures> = Expand<
   >
 >
 
-export function render<TFeature extends RenderFeatures, TTag extends ElementType, TSlot>({
+export function useRender() {
```

Comment:
Pretty sure we can move way more things in here such that everything is simpler and more optimized. But want to do it in small steps.

This essentially wraps the `render(…)` function with the `useMergeRefsFn()` which means every component now gets the `useMergeRefsFn()` for free.